src/vcpkg/base/system.cpp

```diff
@@ -273,6 +273,28 @@ namespace vcpkg
         }();
         return s_home;
     }
+
+    const ExpectedS<Path>& get_system_root() noexcept
+    {
+        static const ExpectedS<Path> s_system_root = []() -> ExpectedS<Path> {
+            auto env = get_environment_variable("SystemRoot");
+            if (const auto p = env.get())
+            {
+                return std::move(*p);
+            }
+            else
+            {
+                return std::string("Expected the SystemRoot environment variable to be always set on Windows.");
+            }
+        }();
+        return s_system_root;
+    }
+
+    const ExpectedS<Path>& get_system32() noexcept
+    {
+        static const ExpectedS<Path> s_system32 = get_system_root().map([](const Path& p) { return p / "system32"; });
```

Comment:
```suggestion
        static const ExpectedS<Path> s_system32 = get_system_root().map([](const Path& p) { return p / "System32"; });
```
I'd prefer it to use the canonical casing (even if it doesn't really matter)